src/vec.rs

```diff
@@ -7,6 +7,35 @@ use core::{
     ops, ptr, slice,
 };
 
+/// Workaround forbidden specialization of Drop
+pub trait VecDrop {
+    fn drop_with_len(&mut self, len: usize);
+}
+
+impl<T> VecDrop for [MaybeUninit<T>] {
+    fn drop_with_len(&mut self, _len: usize) {
+        // Case of a view, drop does nothing
+    }
+}
+
+impl<T, const N: usize> VecDrop for [MaybeUninit<T>; N] {
+    fn drop_with_len(&mut self, len: usize) {
```

Comment:
This should be unsafe, given there is no bounds checking in the implementation.

Reply:
Correct, I made this unsafe.